CRM/Core/BAO/MessageTemplate.php

```diff
@@ -88,7 +90,7 @@ public static function add(&$params) {
           if (!empty($params['workflow_id']) && !CRM_Core_Permission::check('edit system workflow message templates')) {
             throw new \Civi\API\Exception\UnauthorizedException(ts('%1', [1 => $systemWorkflowPermissionDeniedMessage]));
           }
-          elseif (!CRM_Core_Permission::check('edit user-driven message templates')) {
+          if (!CRM_Core_Permission::check('edit user-driven message templates')) {
```

Comment:
elseif is meaningless when previous if throws an exception

Reply:
@eileenmcnaughton i think it might have some meaning, If you have an end user that is able to edit system workflow templates but not the user driven ones (why i have no idea). The change here would mean that the check for user driven message templates permission is now no longer dependant on if you have a workflow_id or not.

Reply:
Right but  if the first IF is true then it throws an exception & never reaches the second  - which  is what elseif achieves when no exception is thrown

Reply:
My point is if the user is able to satisfy the first if completely (we have a workflow id and they have the correct permission) changing from an elseif to an else will mean the 2nd permission check will run no matter if there is a workflow_id in the params or not. With elseif it will only run if there is no workflow id in the params array.

Reply:
So we have 2 checks 

If check 1 is true then an exception is thrown and  the process does not proceed. The elseif is never reached

If not then the second check is performed, as  it  would be if the elseif were there

The second check will still only run if  the first check is false because of the early exit